Tests/TestUtils/XCTestCase+AnyCodableAsserts.swift

```diff
@@ -639,6 +630,51 @@ extension XCTestCase {
 
     // MARK: - Test setup and output helpers
     
+    /// Extracts and returns a set of valid wildcard indexes.
+    ///
+    /// This method only considers keys that match the array access format (ex: `[*123]`).
+    /// It identifies wildcard indexes by:
+    /// 1. Filtering out index values that don't have the wildcard marker `*`.
+    /// 2. Strictly validating the following, and emitting a test failure if any check fails:
+    ///   i. Wildcard character must be placed on the left of the index value (that is, as the leftmost character in the array brackets).
+    ///   ii. The index value must be parsable as a valid `Int` once the wildcard character is removed.
+    ///
+    /// - Parameters:
+    ///   - pathEndKeys: An array of end keys which may contain potential wildcard indexes.
+    ///   - file: The file from which the method is called, used for localized assertion failures.
+    ///   - line: The line from which the method is called, used for localized assertion failures.
+    ///
+    /// - Returns: A set of valid wildcard `Int` indexes.
+    ///
+    /// - Note:
+    ///   Examples of conversions:
+    ///   - `[*123]` -> `123`
+    ///   - `[*ab12]` causes a test failure since "ab12" is not a valid integer.
+    ///   - `[0*]` causes a test failure since the wildcard character `*` is not the leftmost character.
+    private func extractValidWildcardIndexes(pathEndKeys: [String], file: StaticString = #file, line: UInt = #line) -> Set<Int> {
+        let arrayIndexValueRegex = #"^\[(.*?)\]$"#
+        let arrayIndexValues = Set(pathEndKeys
+            .flatMap { getCapturedRegexGroups(text: $0, regexPattern: arrayIndexValueRegex) })
+        
+        let potentialWildcardIndexes = arrayIndexValues
+            .filter { $0.contains("*") }
+        
+        var result: Set<Int> = []
+        for potentialWildcardIndex in potentialWildcardIndexes {
+            if potentialWildcardIndex.first != "*" {
+                XCTFail("TEST ERROR: wildcard indexes must have a single `*` character to the left of the index.", file: file, line: line)
```

Comment:
nit: let's include a simple example for the implementor: `... to the left of the index, ie [*0].`